script.js

```diff
@@ -1,4 +1,100 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+let numOfDice = 0;
+let gameMode = "waiting for number of players";
+let playerTurn = 0;
+let arrayOfPlayers = [];
+
+//store Player's object to be used for sorting diceRolls and leaderboard
+const storePlayerObjects = (userInput) => {
+  for (let i = 0; i < userInput; i += 1) {
+    arrayOfPlayers.push({ id: i + 1, diceRolls: [] });
+  }
+};
+
+//get winner at the end of game , display leaderboard, restart the game
+const findWinner = () => {
+  let sortedScore = arrayOfPlayers.sort((a, b) => b.diceRolls - a.diceRolls);
+  let winner = sortedScore[0].id;
+  let finalMessage = `Winner is player ${winner}.\<br\><br\>\ Leaderboard\<br\>\ ${leaderBoard(
+    arrayOfPlayers
+  )}\<br\><br\>\Type number of players to replay the game.`;
+  resetGame();
+  return finalMessage;
+};
+
+//prep leaderboard
+const leaderBoard = (anArray) =>
+  anArray.reduce(
+    (acc, curVal) =>
+      `${acc} Player ${curVal.id}. Dicerolls: ${curVal.diceRolls}\<br\>\ `,
+    ""
+  );
```

Comment:
Excellent! I like how you used the sort function and decomposed that to a message showcased in order. Very create 👏